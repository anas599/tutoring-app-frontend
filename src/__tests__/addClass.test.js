import '../matchMedia';
import '@testing-library/jest-dom/extend-expect';
import { render, screen } from '@testing-library/react';
import { useDispatch } from 'react-redux';
import ClassForm from '../components/add-class/AddClass';

jest.mock('react-redux', () => ({
  useDispatch: jest.fn(),
}));

describe('ClassForm', () => {
  test('renders the form', () => {
    useDispatch.mockReturnValue(jest.fn());

    const { formContainer } = render(<ClassForm />);

    const tutorNameInput = screen.getByLabelText('Tutor Name');
    const subjectInput = screen.getByLabelText('Subject');
    const durationInput = screen.getByLabelText('Duration');
    const imageInput = screen.getByLabelText('image');

    const submitButton = screen.getAllByText('Add Class');

    expect(tutorNameInput).toBeInTheDocument();
    expect(subjectInput).toBeInTheDocument();
    expect(durationInput).toBeInTheDocument();
    expect(imageInput).toBeInTheDocument();
    expect(submitButton.length).toBeGreaterThan(0);

    expect(formContainer).toMatchSnapshot();
  });
});
