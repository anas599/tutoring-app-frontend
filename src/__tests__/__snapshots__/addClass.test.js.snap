// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClassForm renders the form 1`] = `undefined`;
